### PR TITLE
[DEV-728] Setup GitHub workflow and update documentation

### PR DESCRIPTION
**Pull request by RashikaKarki**
This PR contains updates in documentation and adds setup of GitHub actions to run test. Also, added Rubocop.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (erinosher) [inline CONTRIBUTING.md]: This should probably specify that we follow [Semver](https://semver.org/)

File: CONTRIBUTING.md
Comment (erinosher) [inline CONTRIBUTING.md]: I'm inclined to say this should probably live in an `example` directory. But happy to defer to @theycallmeswift's take

File: README.md
Comment (erinosher) [inline README.md]:
```suggestion
[![Test](https://github.com/MLH/omniauth-mlh/actions/workflows/test.yml/badge.svg)](https://github.com/MLH/omniauth-mlh/actions/workflows/test.yml)
```

This should use the GitHub Actions badge not Travis CI

**Review by erinosher — COMMENTED** [review]
Left a couple of small Notes, but otherwise looks good to me.

File: .github/workflows/test.yml
Comment (theycallmeswift) [inline .github/workflows/test.yml]: Should we test this on a few different versions of Ruby? Specifically 2.7 and the latest 3.x make sense to me

File: CONTRIBUTING.md
Comment (theycallmeswift) [reply CONTRIBUTING.md]: We should remove this section anyway. It's not on the contributor to maintain the version number, that's on the maintainer (us). I do think we may want to document the release process somewhere though. Maybe in the Engineering gitbook?

Also, this is missing the tagging steps:

```
$ git commit -m "vX.Y.Z"
$ git tag -a vX.Y.Z -m "vX.Y.Z"
$ git push origin main --tags
```

File: CONTRIBUTING.md
Comment (theycallmeswift) [reply CONTRIBUTING.md]: We could also generalize this the way this Gem does: https://github.com/auth0/omniauth-auth0/blob/master/CONTRIBUTING.md

Agree that we should avoid linking to a random gist

File: omniauth-mlh.gemspec
Comment (theycallmeswift) [inline omniauth-mlh.gemspec]: 'Major League Hacking (MLH)'

**Review by theycallmeswift — COMMENTED** [review]
We should also add some kind of rubocop linting to ensure code quality / formatting. Otherwise LGTM aside from comments!

File: CONTRIBUTING.md
Comment (RashikaKarki) [reply CONTRIBUTING.md]: I agree. For now I will take this out and keep a backup with tagging step for the Engineering gitbook or any other alternative knowledge base.

File: .github/workflows/test.yml
Comment (RashikaKarki) [reply .github/workflows/test.yml]: Sounds good to me. I will add another version too.

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml:7]: Can you add this comment above this line?

```
# Ruby 2.7 is the MINIMUM Ruby version supported

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml]: I prefer:

```
Metrics/BlockLength:
  CountComments: true 
  Max: 25
  Exclude:
    - "spec/**/*"
```

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml#L1514

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml#L1503

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml#L3912

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml]: This is enabled by default: https://github.com/rubocop/rubocop/blob/master/config/default.yml#L2195

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml]: Make this comment specific to the reason you're overriding the default behavior, not a description of the default. E.g. Answer the question **why** are you excluding `lib/omniauth-mlh.rb`?

File: Rakefile
Comment (theycallmeswift) [inline Rakefile]: You don't need the block here at all. Rubocop will pick up the `.rubocop.yml` file by default and will include all lintable files. You're actually missing some files with your patterns definiton.

File: Rakefile
Comment (theycallmeswift) [inline Rakefile:12]: `task default: [:rubocop, :spec]`

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml:5]: I think you'll address a lot of this in the ticket for adding Rubocop to OHQ, but here are some good resources:

- https://github.com/thoughtbot/guides/blob/main/ruby/.rubocop.yml
- https://github.com/rubocop/rubocop/blob/master/config/default.yml

**Review by theycallmeswift — CHANGES_REQUESTED** [review]
Some more minor feedback, mostly related to rubocop

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml]: I actually dislike this rule. I think we should disable it. This is a good default to include in OHQ as well!

```
RSpec/MultipleExpectations:
  Description: Allow tests to contain multiple expectations
  Enabled: false
```

File: Rakefile
Comment (theycallmeswift) [inline Rakefile]: This is the thing we are disabling in OHQ. We should disable it here as well

**Review by theycallmeswift — APPROVED** [review]
Overall this is about ready to merge. I would include the new "default" rules you're writing for the OHQ Rubocop config (using array notation instead of `%i`, etc) and I also found a good Rspec rule that we can disable here and also include in our new OHQ config!

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml:13]: You're missing the `Description` and `Enabled` attributes for clarity.

File: .rubocop.yml
Comment (theycallmeswift) [inline .rubocop.yml:23]: You're missing the Description and Enabled attributes for clarity.

File: .rubocop.yml
Comment (theycallmeswift) [reply .rubocop.yml]: This was never resolved

File: Rakefile
Comment (theycallmeswift) [reply Rakefile]: This isn't resolved

File: spec/support/shared_examples.rb
Comment (theycallmeswift) [inline spec/support/shared_examples.rb:5]: This file still has `%i[ ]` bracket notation

**Review by theycallmeswift — CHANGES_REQUESTED** [review]

**Review by theycallmeswift — APPROVED** [review]
LGTM!

**Pull request merged**
